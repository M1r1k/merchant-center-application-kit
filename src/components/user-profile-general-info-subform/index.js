export { default } from './user-profile-general-info-subform';
