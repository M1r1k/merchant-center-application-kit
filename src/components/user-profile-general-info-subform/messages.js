import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'UserProfile.GeneralInfoPanel.title',
    description: 'Title of the User Profile general info panel',
    defaultMessage: 'General Information',
  },
  firstName: {
    id: 'UserProfile.GeneralInfoPanel.firstName',
    description: 'First name label',
    defaultMessage: 'First Name',
  },
  lastName: {
    id: 'UserProfile.GeneralInfoPanel.lastName',
    description: 'Last name label',
    defaultMessage: 'Last Name',
  },
  email: {
    id: 'UserProfile.GeneralInfoPanel.email',
    description: 'E-mail label',
    defaultMessage: 'E-mail',
  },
});
