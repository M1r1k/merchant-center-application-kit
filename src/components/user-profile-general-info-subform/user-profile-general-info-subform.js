import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import TextInput from '@commercetools-local/ui-kit/inputs/text-input';
import ValidationError from '@commercetools-local/core/components/validation-error';
import Spacings from '@commercetools-local/ui-kit/materials/spacings';
import CollapsiblePanel from '@commercetools-local/ui-kit/panels/collapsible-panel';
import ErrorMessage from '@commercetools-local/ui-kit/messages/error-message';
import { messages as validationMessages } from '@commercetools-local/utils/validation';
import FormBox from '@commercetools-local/core/components/form-box';
import LabelField from '@commercetools-local/core/components/fields/label-field';
import messages from './messages';

export const UserProfileGeneralInfoSubform = props => (
  <CollapsiblePanel label={<FormattedMessage {...messages.title} />}>
    <FormBox>
      <LabelField
        title={<FormattedMessage {...messages.firstName} />}
        isRequired={true}
      />
      <Spacings.Stack>
        <TextInput
          name="firstName"
          value={props.formik.values.firstName}
          hasError={Boolean(
            props.formik.touched.firstName && props.formik.errors.firstName
          )}
          onChange={props.formik.handleChange}
          isDisabled={props.formik.isSubmitting}
        />
        <ValidationError.Switch
          errors={props.formik.errors.firstName}
          isTouched={props.formik.touched.firstName}
        >
          <ValidationError.Match rule="required">
            <ErrorMessage>
              <FormattedMessage {...validationMessages.required} />
            </ErrorMessage>
          </ValidationError.Match>
        </ValidationError.Switch>
      </Spacings.Stack>
    </FormBox>
    <FormBox>
      <LabelField
        title={<FormattedMessage {...messages.lastName} />}
        isRequired={true}
      />
      <Spacings.Stack>
        <TextInput
          name="lastName"
          value={props.formik.values.lastName}
          hasError={Boolean(
            props.formik.touched.lastName && props.formik.errors.lastName
          )}
          onChange={props.formik.handleChange}
          isDisabled={props.formik.isSubmitting}
        />
        <ValidationError.Switch
          errors={props.formik.errors.lastName}
          isTouched={props.formik.touched.lastName}
        >
          <ValidationError.Match rule="required">
            <ErrorMessage>
              <FormattedMessage {...validationMessages.required} />
            </ErrorMessage>
          </ValidationError.Match>
        </ValidationError.Switch>
      </Spacings.Stack>
    </FormBox>
    <FormBox>
      <LabelField title={<FormattedMessage {...messages.email} />} />
      <TextInput
        name="email"
        value={props.formik.values.email}
        isDisabled={true}
        onChange={props.formik.handleChange}
      />
    </FormBox>
  </CollapsiblePanel>
);
UserProfileGeneralInfoSubform.displayName = 'UserProfileGeneralInfoSubform';
UserProfileGeneralInfoSubform.propTypes = {
  formik: PropTypes.shape({
    isSubmitting: PropTypes.bool.isRequired,
    errors: PropTypes.shape({
      firstName: PropTypes.shape({
        required: PropTypes.bool.isRequired,
      }),
      lastName: PropTypes.shape({
        required: PropTypes.bool.isRequired,
      }),
    }),
    values: PropTypes.shape({
      firstName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
    }),
    touched: PropTypes.shape({
      firstName: PropTypes.bool,
      lastName: PropTypes.bool,
    }),
    handleChange: PropTypes.func.isRequired,
  }),
};

export default UserProfileGeneralInfoSubform;
