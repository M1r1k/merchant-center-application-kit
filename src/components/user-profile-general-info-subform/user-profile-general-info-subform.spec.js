import React from 'react';
import { shallow } from 'enzyme';
import { intlMock } from '@commercetools-local/test-utils';
import { UserProfileGeneralInfoSubform } from './user-profile-general-info-subform';

const createFormikProps = props => ({
  isSubmitting: false,
  values: {
    email: 'john@snow.got',
    firstName: 'John',
    lastName: 'Snow',
  },
  errors: {},
  touched: {},
  handleChange: jest.fn(),

  ...props,
});
const createTestProps = props => ({
  formik: createFormikProps(),

  intl: intlMock,

  ...props,
});

describe('rendering', () => {
  describe('when there are no errors', () => {
    let props;
    let wrapper;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<UserProfileGeneralInfoSubform {...props} />);
    });
    it('should ensure layout structure', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render input field for firstName', () => {
      expect(wrapper).toRender({ name: 'firstName' });
    });
    it('should render input field for lastName', () => {
      expect(wrapper).toRender({ name: 'lastName' });
    });
    it('should render input field for email', () => {
      expect(wrapper).toRender({ name: 'email' });
    });
  });

  describe('when submitting', () => {
    let props;
    let wrapper;
    beforeEach(() => {
      props = createTestProps({
        formik: createFormikProps({
          isSubmitting: true,
        }),
      });
      wrapper = shallow(<UserProfileGeneralInfoSubform {...props} />);
    });
    it('should disable all input fields', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should disable the first name input', () => {
      expect(wrapper.find({ name: 'firstName' })).toHaveProp(
        'isDisabled',
        true
      );
    });
  });
});
