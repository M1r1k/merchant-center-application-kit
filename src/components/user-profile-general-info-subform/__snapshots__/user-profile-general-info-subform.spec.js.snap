// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering when submitting should disable all input fields 1`] = `
<CollapsiblePanel
  label="UserProfile.GeneralInfoPanel.title"
>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={true}
      title="UserProfile.GeneralInfoPanel.firstName"
    />
    <TextInput
      isDisabled={true}
      name="firstName"
      onChange={[MockFunction]}
      value="John"
    />
  </FormBox>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={true}
      title="UserProfile.GeneralInfoPanel.lastName"
    />
    <TextInput
      isDisabled={true}
      name="lastName"
      onChange={[MockFunction]}
      value="Snow"
    />
  </FormBox>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={false}
      title="UserProfile.GeneralInfoPanel.email"
    />
    <TextInput
      isDisabled={true}
      name="email"
      onChange={[MockFunction]}
      value="john@snow.got"
    />
  </FormBox>
</CollapsiblePanel>
`;

exports[`rendering when there are no errors should ensure layout structure 1`] = `
<CollapsiblePanel
  label="UserProfile.GeneralInfoPanel.title"
>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={true}
      title="UserProfile.GeneralInfoPanel.firstName"
    />
    <TextInput
      isDisabled={false}
      name="firstName"
      onChange={[MockFunction]}
      value="John"
    />
  </FormBox>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={true}
      title="UserProfile.GeneralInfoPanel.lastName"
    />
    <TextInput
      isDisabled={false}
      name="lastName"
      onChange={[MockFunction]}
      value="Snow"
    />
  </FormBox>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={false}
      title="UserProfile.GeneralInfoPanel.email"
    />
    <TextInput
      isDisabled={true}
      name="email"
      onChange={[MockFunction]}
      value="john@snow.got"
    />
  </FormBox>
</CollapsiblePanel>
`;
